### DISCUSSION TITLE
Display LILYGO T5 ePaper 4.7 with LVGL not working

### DISCUSSION THREAD

**Discussion by Jason2866** [arendst/Tasmota#13473] [Q&A] [answered]
@gemu2015 Can you help?

user_config_override.h 
```
#ifdef FIRMWARE_TTGO47
    // -- CODE_IMAGE_STR is the name shown between brackets on the
    //    Information page or in INFO MQTT messages
    #undef CODE_IMAGE_STR
    #define CODE_IMAGE_STR "ttgo47"

#define USE_DISPLAY
#define USE_LVGL
#define USE_LILYGO47
#define SHOW_SPLASH

#endif // #ifdef FIRMWARE_TTGO47_DISPLAY
```
env entry added in Platformio
```
[env:tasmota32-ttgo47]
extends                 = env:tasmota32idf4
board                   = esp32_16M
upload_port             = /dev/cu.SLAB_USBtoUART
build_unflags           = ${env:tasmota32idf4.build_unflags}
build_flags             = ${env:tasmota32idf4.build_flags}
                          -DFIRMWARE_TTGO47
lib_extra_dirs          = lib/libesp32
                          lib/lib_basic
                          lib/lib_i2c
                          lib/lib_display
                          lib/libesp32_lvgl
                          lib/libesp32_epdiy
```
Compile fails with the Touch Driver

**Comment by gemu2015** [comment, marked as answer]
add
#define USE_TOUCH_BUTTONS

and it should compile.

however since i do not have a display with touch i could not test functionality

be warned with that display!
never expose it to sun or ultraviolet light. i ruined mine when i let it stay near a window with partial sunlight.
i got heavy and unrecoverable burn ins

**Reply by blakadder** [reply]
The define is required even without the touch pad addon?

**Reply by gemu2015** [reply]
no normally not required, only if you have the touch panel.
i though jason got compile errors missing touch?

**Reply by Jason2866** [reply]
I have to add the definition to compile successfully. I have NO touchpad.
Without the touchpad stuff i didnt managed to get it compiled with LVGL

**Reply by gemu2015** [reply]
i now entered your config and could reproduce your error.
#ifdef FIRMWARE_TTGO47
causes this behavior

did not dig into it further but my config works

using 
#define device_EPD47

#ifdef device_EPD47
.....
#endif

**Reply by gemu2015** [reply]
had a corrupted build. removed cache and pio and now everything compiles with your setup and no TOUCH_BUTTONS
defined

**Reply by Jason2866** [reply]
Indeed it compiles, weird behaviour of platformio (once again).

**Comment by gemu2015** [comment]
tasmota epd47 does NOT use the TTGO driver but a different one which supports the EPD47 board
https://github.com/vroland/epdiy
it has its own waveform drivers.
and you can use edp47 for LVGL and BERRY since it also use the renderer class.

**Comment by Jason2866** [comment]
TTGO released the specific waveform definition. https://github.com/Xinyuan-LilyGO/LilyGo-EPD47/tree/master/Waveform_header
Could you integrate?

**Reply by gemu2015** [reply]
download my fork, i integrated the TTGO waveform.
i can see no difference in function on the first view.
